### PR TITLE
feat: separate commands for solo node delete

### LINKED ISSUE #529: Reusable components of `solo node delete` to be leveraged for SDK testing
Please update and leverage this design doc to assist:  [Solo Reusability for DAB Test Cases (DRAFT)](https://www.notion.so/swirldslabs/Solo-Reusability-for-DAB-Test-Cases-DRAFT-404c502226384b449878ccfe089587df?pvs=6&utm_content=404c5022-2638-4b44-9878-ccfe089587df&utm_campaign=T035PD799T9&n=slack&n=slack_link_unfurl)
relates to https://github.com/hashgraph/solo/issues/522

depends on https://github.com/hashgraph/solo/pull/499 to prevent merge conflicts

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
## Description

- Separates the flow of `solo node delete` into 3 new commands that can be executed separately and one after the other: `solo node delete-prepare`, `solo node delete-submit-transactions`, `solo node delete-execute`.

The behaviour of the new commands is as follows:

```
solo node delete-prepare  --node-id node2 --release-tag v0.54.0-alpha.4 --namespace solo-e2e --output-dir context
---------
Initialize
Identify existing network nodes
Load node admin key
Prepare upgrade zip file for node upgrade process
Check existing nodes staked amount
Save context data

```
Saves data from the `ctx` object to the specified directory by `--output-dir`

```
solo node delete-submit-transactions --namespace solo-e2e --input-dir context
---------
Initialize
Load context data
Send node delete transaction
Send prepare upgrade transaction
Send freeze upgrade transaction
```
Loads data into the `ctx` object from the specified directory by `--input-dir`

```
solo node delete-execute --namespace solo-e2e --input-dir context [--local-build-path xxxxxxx]
---------


Initialize 
Load context data
Download generated files from an existing node 
Prepare staging directory 
  Copy Gossip keys to staging
  Copy gRPC TLS keys to staging
Copy node keys to secrets 
  Copy TLS keys 
  Node: node1 
    Copy Gossip keys 
  Node: node3 
    Copy Gossip keys 
Check network nodes are frozen 
  Check network pod: node1  
  Check network pod: node2  
  Check network pod: node3  
Get node logs and configs 
Update chart to use new configMap 
Kill nodes to pick up updated configMaps 
Check node pods are running 
  Check Node: node1 
  Check Node: node3
Fetch platform software into all network nodes 
  Update node: node1 
  Update node: node3 
Setup network nodes 
  Node: node1 
    Set file permissions 
  Node: node3 
    Set file permissions 
Start network nodes 
  Start node: node1 
  Start node: node3 
Enable port forwarding for JVM debugger
Check all nodes are ACTIVE 
  Check network pod: node1  
  Check network pod: node3  
Check all node proxies are ACTIVE 
  Check proxy for node: node1 
  Check proxy for node: node3 
Trigger stake weight calculate 
Finalize

```
Loads data into the `ctx` object from the specified directory by `--input-dir`

### Related Issues

* Closes #529

### CODE REVIEW

**Review by jeromy-cannon — CHANGES_REQUESTED** [review]
see comments

File: .github/workflows/zxc-code-analysis.yaml [hunk 1]
```diff
@@ -268,26 +278,19 @@ jobs:
...
-          pattern: "e2e_test_report_*"
-          path: "e2e_test_report"
-
-      - name: Publish E2E Test Report
-        uses: EnricoMi/publish-unit-test-result-action@82082dac68ad6a19d980f8ce817e108b9f496c2a # v2.17.1
-        if: ${{ (inputs.enable-codecov-analysis || inputs.enable-codacy-coverage) && inputs.enable-e2e-coverage-report && !cancelled() && !failure() }}
-        with:
-          check_name: "E2E Test Report"
-          files: "e2e_test_report/**/*.xml"
+          name: ${{ inputs.e2e-relay-coverage-report }}
```
Comment (JeffreyDallas) [inline .github/workflows/zxc-code-analysis.yaml:292]: This step was accidentally deleted, need to revert

File: src/commands/node.mjs [hunk 2]
```diff
@@ -2584,6 +2548,57 @@ export class NodeCommand extends BaseCommand {
               })
             }
           })
+          .command({
+            command: 'delete-prepare',
+            desc: 'Prepares the deletion of a node with a specific version of Hedera platform',
+            builder: y => flags.setCommandFlags(y, ...NodeCommand.DELETE_PREPARE_FLAGS_LIST.concat(flags.nodeID)),
+            handler: argv => {
+              nodeCmd.logger.debug('==== Running \'node add\' ===')
```
Comment (JeffreyDallas) [inline src/commands/node.mjs:2556]: should be Running \'node delete-prepare\'

File: src/commands/node.mjs [hunk 3]
```diff
@@ -2584,6 +2548,57 @@ export class NodeCommand extends BaseCommand {
...
+                process.exit(1)
+              })
+            }
+          })
+          .command({
+            command: 'delete-submit-transactions',
+            desc: 'Submits transactions to the network nodes for deleting a node',
+            builder: y => flags.setCommandFlags(y, ...NodeCommand.DELETE_SUBMIT_TRANSACTIONS_FLAGS_LIST),
+            handler: argv => {
+              nodeCmd.logger.debug('==== Running \'node add\' ===')
```
Comment (JeffreyDallas) [inline src/commands/node.mjs:2573]: should be Running \'node delete-submit-transactions\'

File: src/commands/node.mjs [hunk 4]
```diff
@@ -2584,6 +2548,57 @@ export class NodeCommand extends BaseCommand {
...
+                process.exit(1)
+              })
+            }
+          })
+          .command({
+            command: 'delete-execute',
+            desc: 'Executes the deletion of a previously prepared node',
+            builder: y => flags.setCommandFlags(y, ...NodeCommand.DELETE_EXECUTE_FLAGS_LIST),
+            handler: argv => {
+              nodeCmd.logger.debug('==== Running \'node add\' ===')
```
Comment (JeffreyDallas) [inline src/commands/node.mjs:2590]: should be Running 'node delete-execute'

File: .github/workflows/flow-pull-request-checks.yaml [hunk 5]
```diff
@@ -52,16 +52,6 @@ jobs:
...
-  update-readme:
-    name: "Update README.md"
-    uses: ./.github/workflows/flow-update-readme.yaml
-    with:
-      commit-changes: false
-    secrets:
-      GH_ACCESS_TOKEN: ${{ secrets.GH_ACCESS_TOKEN }}
-      GH_ACCESS_GPG_KEY: ${{ secrets.GH_ACCESS_GPG_KEY }}
-      GH_ACCESS_PASSPHRASE: ${{ secrets.GH_ACCESS_PASSPHRASE }}
-
```
[the remark this answers is not in the document]
Comment (JeffreyDallas) [reply .github/workflows/flow-pull-request-checks.yaml]: Fixed

File: .github/workflows/zxc-code-analysis.yaml (hunk 1 above)
Comment (JeffreyDallas) [reply .github/workflows/zxc-code-analysis.yaml:292]: Fixed

**Review by jeromy-cannon — APPROVED** [review]

**Pull request merged**